internal/types/document.go

```diff
@@ -167,6 +172,55 @@ func (d *Document) Keys() []string {
 	return keys
 }
 
+// Values returns a copy of document's values in the same order as Keys().
+//
+// If document or document's fields are not set (nil), it returns nil.
+func (d *Document) Values() []any {
+	if d == nil || d.fields == nil {
+		return nil
+	}
+
+	values := make([]any, len(d.fields))
+	for i, field := range d.fields {
+		values[i] = field.value
+	}
+
+	return values
+}
+
+// FindDuplicateKey returns the first duplicate key in the document and true if duplicate exists.
+// If duplicate keys don't exist it returns empty string and false.
+func (d *Document) FindDuplicateKey() (string, bool) {
+	seen := make(map[string]struct{}, len(d.fields))
+	for _, field := range d.fields {
+		if _, ok := seen[field.key]; ok {
+			return field.key, true
+		}
+
+		seen[field.key] = struct{}{}
+	}
+
+	return "", false
+}
+
+// RemoveDuplicateKeys removes duplicate keys from the document. The last value of each duplicate key is kept.
+func (d *Document) RemoveDuplicateKeys() {
```

Comment:
We should not have that method at all.

Instead, we should validate documents in handler and return error, not silently remove duplicates.

`types.Document.ValidateData()` should already do that

Reply:
(yes, even if MongoDB works this way… Doing the same feels like a violation of common sense)